### PR TITLE
Compressor Head Complete App

### PR DESCRIPTION
**Pull request by SachinRammoorthy**
compressor-head-android with complete functionality. Generates a link to the compressed image, has image preview, has option for constrain proportions, height, width, and format. Also has a placeholder image for when the url is not typed in yet, and an error image if the url is invalid.

Task: https://codein.withgoogle.com/dashboard/task-instances/6688682775937024/

screen shot 2017-12-10 at 10 35 37 am
screenshot_2017-12-10-10-36-22

### CODE REVIEW

**Comment by garg-anuj** [comment]
https://travis-ci.org/jboss-outreach/compressor-head-android/builds/314231030#L1545

**Comment by SachinRammoorthy** [comment]
@garg-anuj travis-ci check passed

**Comment by SachinRammoorthy** [comment]
Also, image preview is being compressed before display.

File: app/build.gradle [hunk 1]
```diff
@@ -1,25 +1,27 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion rootProject.ext.compileSdkVersion
+compileSdkVersion rootProject.ext.compileSdkVersion
```
Comment (m-murad) [inline app/build.gradle]: Undo this formatting.

File: app/build.gradle [hunk 2]
```diff
@@ -1,25 +1,27 @@
...
     buildToolsVersion rootProject.ext.supportLibVersion
 
     defaultConfig {
         applicationId "com.community.jboss.compressorhead"
-        minSdkVersion 16
+        minSdkVersion 16
```
Comment (m-murad) [inline app/build.gradle]: Undo this change.

File: app/build.gradle [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
-        minSdkVersion 16
+        minSdkVersion 16    
         targetSdkVersion rootProject.ext.compileSdkVersion
         versionCode 1
         versionName "1.0"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
         vectorDrawables.useSupportLibrary = true
     }
+    
+
```
Comment (m-murad) [inline app/build.gradle]: Don't add unnecessary blank lines.

File: app/build.gradle [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
+    
+    
     buildTypes {
         release {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
-
+
```
Comment (m-murad) [inline app/build.gradle]: Undo this

File: app/src/main/java/com/community/jboss/compressorhead/activities/MainActivity.java [hunk 5]
```diff
@@ -1,53 +1,98 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        editTextUrl = (EditText) findViewById(R.id.image_url);
+        editTextWidth = (EditText) findViewById(R.id.image_width);
+        editTextHeight = (EditText) findViewById(R.id.image_height);
+        imageView = (ImageView) findViewById(R.id.image_view);
+        textView = (TextView) findViewById(R.id.text_view);
+        spinner =(Spinner) findViewById(R.id.spinner);
+
```
Comment (m-murad) [inline app/src/main/java/com/community/jboss/compressorhead/activities/MainActivity.java]: Again don't add blank lines.

File: app/src/main/java/com/community/jboss/compressorhead/activities/MainActivity.java [hunk 6]
```diff
@@ -1,53 +1,98 @@
...
     }
 
-    @Override
-    public boolean onCreateOptionsMenu(Menu menu) {
-        final MenuInflater inflater = getMenuInflater();
-        inflater.inflate(R.menu.menu_main, menu);
-        return true;
+
+    public void generatePreview(View view) {
+        imageUri = editTextUrl.getText().toString();
```
Comment (m-murad) [inline app/src/main/java/com/community/jboss/compressorhead/activities/MainActivity.java]: add `.trim()` here.

File: app/src/main/java/com/community/jboss/compressorhead/activities/MainActivity.java [hunk 7]
```diff
@@ -1,53 +1,98 @@
...
+             format = "jpeg";
+        }
+
+        generateUrl();
     }
+
+
+    public void generateUrl() {
+
+            textView.setText("http://compressor-head.appspot.com/image/?image_url=" + imageUri +
```
Comment (m-murad) [inline app/src/main/java/com/community/jboss/compressorhead/activities/MainActivity.java]: Use the value returned by `getURL()` method of `CompressorHead` method here? What is the need to build the URL again.

File: app/src/main/java/com/community/jboss/compressorhead/activities/MainActivity.java [hunk 8]
```diff
@@ -1,53 +1,98 @@
...
-        final MenuInflater inflater = getMenuInflater();
-        inflater.inflate(R.menu.menu_main, menu);
-        return true;
+
+    public void generatePreview(View view) {
+        imageUri = editTextUrl.getText().toString();
+        imageWidth = Integer.parseInt(editTextWidth.getText().toString());
+        imageHeight = Integer.parseInt(editTextHeight.getText().toString());
+
+        Picasso.with(getApplicationContext()).load("http://compressor-head.appspot.com/image/?image_url=" + imageUri +
```
Comment (m-murad) [inline app/src/main/java/com/community/jboss/compressorhead/activities/MainActivity.java]: Instead of building the URL here use the value returned by `getURL()` function of `CompressorHead` class here.

**Comment by garg-anuj** [comment]
Thanks  for your awesome work.
